src/main/java/com/atlassian/jira/cloud/jenkins/config/ConfigManagementLink.java

```diff
@@ -124,7 +123,7 @@ public String getRegexFromFormData(final JSONObject formData, final String key)
     }
 
     @VisibleForTesting
-    private void sendConfigDataToJira(final JSONObject formData) {
+    private void sendConfigDataToJira(final JSONObject formData) throws Exception {
```

Comment:
Spent some time deciding on the error type, opinions welcome.
most likely it will be a Unauthorized error of some shape, but potentially a IOException as well, but ultimately want to treat all errors the same....

Reply:
Maybe create a custom exception called something like JiraConnectionException? Throwing a generic exception on a method signature doesn't provide much in the way of clarity. Throwing a JiraConnectionException that points to a custom exception with some Javadoc comments might help future devs who have the misfortune of working in this codecase :laughcry: And if we wrap the `this.pluginConfigApi.sendConnectionData(...` block in a try catch, we can get a little more specific with something like:

```
try stuffs...
} catch (IOException | InterruptedException | UnauthorizedException e) {
            throw new JiraConnectionException("Connection failed for site: " + siteName, e);
} catch (Exception e) {
    // Handle unexpected exceptions in here
}

Reply:
its thrown down below in the catch(adding it to method signiture now!)
I may have misunderstood your original comment tho :\

Reply:
lol me thinks you did. So it looks like you're extending from the custom class now which is good but we still want to catch specific errors. So in the catch block we could have something like 

```
} catch (IOException e) {
    logger.error("Error connecting to Jira for site {}: {}", siteName, e.getMessage(), e);
    throw new JiraConnectionFailedException(
            String.format("Failed to connect to Jira for site '%s'. Error: %s", siteName, e.getMessage()), e);
} catch (UnauthorizedException e) {
    logger.error("Unauthorized access to Jira for site {}: {}", siteName, e.getMessage(), e);
    throw new JiraConnectionFailedException(
            String.format("Unauthorized access to Jira for site '%s'. Error: %s", siteName, e.getMessage()), e);
} catch (Exception e) {
    logger.error("Unexpected error connecting to Jira for site {}: {}", siteName, e.getMessage(), e);
    throw new JiraConnectionFailedException(
            String.format("Unexpected error for site '%s'. Error: %s", siteName, e.getMessage()), e);
}
```